samples/hybrid-connections/java/simple-http-demo/src/main/java/HttpListener.java

```diff
@@ -0,0 +1,66 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Scanner;
+
+import com.microsoft.azure.relay.HybridConnectionListener;
+import com.microsoft.azure.relay.RelayConnectionStringBuilder;
+import com.microsoft.azure.relay.RelayedHttpListenerResponse;
+import com.microsoft.azure.relay.TokenProvider;
+
+public class HttpListener {
+	static final String CONNECTION_STRING_ENV_VARIABLE_NAME = "RELAY_CONNECTION_STRING";
+	static final RelayConnectionStringBuilder connectionParams = new RelayConnectionStringBuilder(
+			System.getenv(CONNECTION_STRING_ENV_VARIABLE_NAME));
+
+	public static void main(String[] args) throws URISyntaxException {
+		TokenProvider tokenProvider = TokenProvider.createSharedAccessSignatureTokenProvider(
+				connectionParams.getSharedAccessKeyName(),
+				connectionParams.getSharedAccessKey());
+		HybridConnectionListener listener = new HybridConnectionListener(
+				new URI(connectionParams.getEndpoint().toString() + connectionParams.getEntityPath()),
+				tokenProvider);
+
+		// The "context" object encapsulates both the incoming request and the outgoing response
+		listener.setRequestHandler((context) -> {
+			String receivedText = "";
+			if (context.getRequest().getInputStream() != null) {
+				try (BufferedReader reader = new BufferedReader(new InputStreamReader(context.getRequest().getInputStream(), "UTF8"))) {
+					StringBuilder builder = new StringBuilder();
+					String inputLine;
+					while ((inputLine = reader.readLine()) != null) {
+						builder.append(inputLine);
```

Comment:
>inputLine [](start = 21, length = 9)

Out of curiosity if the input stream had multiple lines in it does this strip out any \r and \n characters?

Reply:
it does strip out the \r nd \n chars

---
In reply to: [267938967](https://github.com/Azure/azure-relay/pull/41#discussion_r267938967) [](ancestors = 267938967)